src/app/components/request-password.tsx

```diff
@@ -0,0 +1,93 @@
+import { useState, useCallback, FormEvent } from 'react';
+import { Box, color, Input, Stack } from '@stacks/ui';
+import UnlockSession from '@assets/images/unlock-session.png';
+import { Text } from '@app/components/typography';
+import { PageTitle } from '@app/components/page-title';
+import { useWaitingMessage, WaitingMessages } from '@app/common/utils/use-waiting-message';
+import { useWallet } from '@app/common/hooks/use-wallet';
+import { useAnalytics } from '@app/common/hooks/analytics/use-analytics';
+import { SettingsSelectors } from '@tests/integration/settings.selectors';
+import { buildEnterKeyEvent } from './link';
+import { ErrorLabel } from './error-label';
+import { PrimaryButton } from './primary-button';
+
+const waitingMessages: WaitingMessages = {
+  '2': 'Verifying password…',
+  '10': 'Still working…',
+  '20': 'Almost there.',
+};
+
+interface RequestPasswordProps {
+  onSuccess: (password: string) => void;
```

Comment:
```suggestion
  onSuccess(password: string): void;
```

favour shorthand fn type notation

Reply:
Agree.